### PR TITLE
Cairo runner new unit tests

### PR DESCRIPTION
**Pull request by daphneherlambda**
This PR add test to ecdsa.rs file

### CODE REVIEW

File: src/cairo_runner.rs [hunk 1]
```diff
@@ -1140,6 +1140,21 @@ mod test {
...
+        let program = fs::read_to_string(path).unwrap();
+        let mut runner =
+            PyCairoRunner::new(program, Some("main".to_string()), None, false).unwrap();
+        runner
+            .cairo_run_py(false, None, None, None, None, None)
+            .unwrap();
+        Python::with_gil(|py| match runner.get_segment_used_size(100, py) {
+            Ok(v) => assert!(false, "get segment used size with invalid data: {v:?}"),
+            Err(e) => assert!(true),
+        });
```
Comment (fmoletta) [inline src/cairo_runner.rs:1155]:
```suggestion
        Python::with_gil(|py| assert!(runner.get_segment_used_size(100, py).is_err()) 
        );
```

File: src/cairo_runner.rs [hunk 2]
```diff
@@ -1497,6 +1512,24 @@ mod test {
...
+            program,
+            Some("main".to_string()),
+            Some(String::from("all")),
+            false,
+        )
+        .unwrap();
+
+        match runner.initial_fp() {
+            Ok(v) => assert!(false, "get initial fp with invalid data error: {v:?}"),
+            Err(_e) => assert!(true),
```
Comment (fmoletta) [inline src/cairo_runner.rs:1529]:
```suggestion
        assert!(runner.initial_fp().is_err())
```

File: src/ecdsa.rs [hunk 3]
```diff
@@ -55,3 +55,144 @@ impl ToPyObject for PySignature {
...
+            Some("main".to_string()),
+            Some("all".to_string()),
+            false,
+        )
+        .unwrap();
+
+        match runner.initialize() {
+            Ok(_v) => assert!(true),
+            Err(e) => assert!(false, "error: {e:?}"),
+        }
```
Comment (fmoletta) [inline src/ecdsa.rs:121]: We could use unwrap or expect here
```suggestion
        runner.initialize().expect("Failed to initialize VM")
```

File: src/ecdsa.rs [hunk 4]
```diff
@@ -55,3 +55,144 @@ impl ToPyObject for PySignature {
...
+
+        let mut binding = runner.pyvm.vm.borrow_mut();
+        let signature_builtin = binding.get_signature_builtin().unwrap();
+
+        match signature.update_signature(signature_builtin) {
+            Ok(_v) => assert!(true),
+            Err(e) => assert!(false, "error: {e:?}"),
+        }
```
Comment (fmoletta) [inline src/ecdsa.rs:129]:
```suggestion
        assert_eq!(signature.update_signature(signature_builtin), Ok(()));
```

File: src/ecdsa.rs [hunk 5]
```diff
@@ -55,3 +55,144 @@ impl ToPyObject for PySignature {
...
+            Some("main".to_string()),
+            Some("all".to_string()),
+            false,
+        )
+        .unwrap();
+
+        match runner.initialize() {
+            Ok(_v) => assert!(true),
+            Err(e) => assert!(false, "error: {e:?}"),
+        }
```
Comment (fmoletta) [inline src/ecdsa.rs:163]:
```suggestion
        runner.initialize().expect("Failed to initialize VM")
```

File: src/cairo_runner.rs (hunk 1 above)
Comment (daphneherlambda) [reply src/cairo_runner.rs]: done!

File: src/cairo_runner.rs (hunk 2 above)
Comment (daphneherlambda) [reply src/cairo_runner.rs]: done!

File: src/ecdsa.rs (hunk 3 above)
Comment (daphneherlambda) [reply src/ecdsa.rs]: done!

File: src/ecdsa.rs (hunk 4 above)
Comment (daphneherlambda) [reply src/ecdsa.rs]: done!

**Review by fmoletta — APPROVED** [review]

File: src/lib.rs [hunk 6]
```diff
@@ -20,3 +20,21 @@ fn cairo_rs_py(_py: Python, m: &PyModule) -> PyResult<()> {
...
+
+    #[test]
+    fn cairo_rs_py_test() {
+        Python::with_gil(|py| {
+            let module = PyModule::new(py, "My Module");
+            let result = crate::cairo_rs_py(py, module.unwrap());
+            match result {
+                Ok(_v) => assert!(true),
+                Err(e) => assert!(false, "error: {e:?}"),
+            }
```
[the remark this answers is not in the document]
Comment (daphneherlambda) [reply src/lib.rs]: uuu I missed this one, fixed now!

**Pull request merged**